src/app/input-bar/input-bar.component.ts

```diff
@@ -37,46 +39,120 @@ export class InputBarComponent implements OnInit {
 
   private parseInput(input: string) {
     const filters: FilterOption<any>[] = [];
-    let groupBy: GroupByKeys = GroupByKeys.None;
+    let grouping: GroupingOptions = {
+      groupBy: GroupByKeys.None,
+      sort: GroupOrders.None
+    };
 
     if (!input) {
       return {
         filters,
-        groupBy
+        grouping
       };
     }
-    const splitInput = input.trim().split(' ');
 
-    for (let i = 0; i < splitInput.length; i++) {
-      if (splitInput[i] !== '{') {
-        const newFilter = this.parseArgument(splitInput[i]);
+    // We put each space seperated token in a stack
+    const stackInput = input.trim().split(' ').reverse();
+
+    let currInputArg: string;
+    while (stackInput.length > 0) {
+      currInputArg = stackInput.pop();
+      if (currInputArg === '{') {
+        grouping = this.parseGrouping(stackInput);
+        if (grouping.groupBy === GroupByKeys.Rewrite) {
+          // An empty inputString tells us to not highlight this regex filter
```

Comment:
This sounds like a hack to me. Should we add a new field instead? (e.g. isFromRewrite)